crates/core/src/matcher.rs

```diff
@@ -43,17 +43,10 @@ pub trait Matcher<L: Language> {
   }
 
   fn match_node<'tree>(&self, node: Node<'tree, L>) -> Option<NodeMatch<'tree, L>> {
-    let mut env = self.get_meta_var_env();
+    // in future we might need to customize initial MetaVarEnv
+    let mut env = MetaVarEnv::new();
     let node = self.match_node_with_env(node, &mut env)?;
-    env.match_constraints().then_some(NodeMatch::new(node, env))
```

Comment:
`constraints` are not a universally useful feature. So I moved it to specific Mathcer implementation.